src/components/Forms/BaseInput.jsx

```diff
@@ -9,7 +10,13 @@ const sizes = {
     large: "px-5 py-3.5 text-md leading-md", // 50px
 };
 
-export const BaseInput = ({ as: Tag, size = "medium", isError, className, isRequired, ...rest }) => {
+export const BaseInput = ({ as: Tag, size = "medium", isError, className, isRequired, value, ...rest }) => {
+    // added regexp to return only non-whitespace characters for strings
+    const stringValue = isString(value) && value.replace(/[^.\S]+/g, "");
+    // Added trim method for string value to prevent leading and trailing spaces.
```

Comment:
there is no trim method used, you can remove if it is not used anymore

Reply:
removed extra comment